tests/ui/test_ui_tools.py

```diff
@@ -1526,21 +1526,18 @@ def test_private_message_to_self(self, mocker):
           ('msg_link', 'Foo'), ' ', ('msg_link_index', '[1]'),
           ('msg_link', 'Bar'), ' ', ('msg_link_index', '[2]')]),
         ('<a href="http://foo">http://foo</a>',
-         [('msg_link', 'http://foo'), ' ', ('msg_link_index', '[1]')]),
+         [('msg_link', 'foo'), ' ', ('msg_link_index', '[1]')]),
```

Comment:
This seems unexpected, but it might just be because this 'domain name' is so short.

This is something of a tricky problem, as I've seen links which are eg. query strings, so the text becomes `some_file?this_thing`, which looks unexpected (and real cases even more so). I'm reconsidering how we should handle them, though I don't see an easy solution straight away.

For now this looks OK.